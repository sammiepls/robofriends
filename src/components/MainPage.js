import React, { Component } from "react";

import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import ErrorBoundry from "./ErrorBoundary";
import Header from "./Header";

export class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots.filter((robot) =>  robot.name.toLowerCase().includes(searchField.toLowerCase())
    );
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          {isPending ? (
            <h1>Loading</h1>
          ) : (
            <ErrorBoundry>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundry>
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;

// Hooks
// import React, {useEffect} from "react";
// import CardList from "./CardList";
// import SearchBox from "./SearchBox";
// import Scroll from "./Scroll";
// import ErrorBoundary from "./ErrorBoundary";
// import Header from "./Header";
// import "../MainPage.css";

// const MainPage = ({ dispatchRequestRobots, robots, onSearchChange, searchField, isPending }) => {

//   useEffect(() => {
//     if (robots.length > 0) return;
//     dispatchRequestRobots();
//   }, [dispatchRequestRobots, robots]);

//   const filteredRobots = robots.filter((robot) => {
//     return robot.name.toLowerCase().includes(searchField.toLowerCase());
//   });

//   return (
//     <div className="App tc">
//       <Header />
//       <SearchBox searchChange={onSearchChange} />
//       <Scroll>
//         {isPending ? (
//           "Is Loading"
//         ) : (
//           <ErrorBoundary>
//             <CardList robots={filteredRobots} />
//           </ErrorBoundary>
//         )}
//       </Scroll>
//     </div>
//   );
// };


// export default MainPage;
