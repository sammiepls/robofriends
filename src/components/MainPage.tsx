// @ts-nocheck
import React, { Component } from "react";

import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import ErrorBoundry from "./ErrorBoundary";
import Header from "./Header";

export class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots.filter((robot) =>  robot.name.toLowerCase().includes(searchField.toLowerCase())
    );
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          {isPending ? (
            <h1>Loading</h1>
          ) : (
            <ErrorBoundry>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundry>
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;

// Hooks
// import React, {useEffect} from "react";
// import CardList from "./CardList";
// import SearchBox from "./SearchBox";
// import Scroll from "./Scroll";
// import ErrorBoundary from "./ErrorBoundary";
// import Header from "./Header";
// import "../MainPage.css";
// import { Robot} from "redux/actions/requestActions";


// interface MainPageProps {
//   dispatchRequestRobots: () => void;
//   robots: Robot[];
//   onSearchChange: (event:React.SyntheticEvent<HTMLInputElement>) => void;
//   searchField: string;
//   isPending: boolean;
// }

// const MainPage:React.FC<MainPageProps> = ({ dispatchRequestRobots, robots, onSearchChange, searchField, isPending }) => {

//   useEffect(() => {
//     if (robots.length > 0) return;
//     dispatchRequestRobots();
//   }, [dispatchRequestRobots, robots]);

//   const filteredRobots = robots.filter((robot:Robot) => {
//     return robot.name.toLowerCase().includes(searchField.toLowerCase());
//   });

//   return (
//     <div className="App tc">
//       <Header />
//       <SearchBox searchChange={onSearchChange} searchField={searchField}/>
//       <Scroll>
//         {isPending ?
//         <>(
//           "Is Loading"
//         )</> : (
//           <ErrorBoundary>
//             <CardList robots={filteredRobots} />
//           </ErrorBoundary>
//         )}
//       </Scroll>
//     </div>
//   );
// };


//export default MainPage;
