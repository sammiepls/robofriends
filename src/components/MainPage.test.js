import React from "react";
import { shallow } from "enzyme";
import MainPage from "./MainPage";

describe("test MainPage", () => {
  let wrapper;
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };

  beforeEach(() => {
    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it("should render the App component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("filters robots correctly", () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: "John",
          email: "john@gmail.com",
        },
        {
          id: 4,
          name: "Alice",
          email: "alice@gmail.com",
        },
      ],
      searchField: "a",
      isPending: false,
    };
    const wrapper2 = shallow(<MainPage {...mockProps2}/>)
    expect(wrapper2.instance().filterRobots()).toEqual([
      {
        id: 4,
        name: "Alice",
        email: "alice@gmail.com",
      },
    ]);
  })
});
