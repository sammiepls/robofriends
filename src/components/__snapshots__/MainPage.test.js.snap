// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MainPage should render the App component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
